### PR TITLE
Merge WebAssembly to Master

### PR DESCRIPTION
**Pull request by Cellule**
This is the first step to bring WebAssembly to the product.
Our implementation is not currently complete, but working directly in master will alleviate some overhead for the people working on WebAssembly and help verify the code's stability through the normal testing pipeline.
These changes didn't cause any noticeable perf regressions when testing in IB and on my machine.

This is a big merge and I expect many parties will have a say on some changes.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/1588)

### CODE REVIEW

**Comment by Cellule** [comment]
@dotnet-bot test slow tests please
@dotnet-bot test linux tests please
@dotnet-bot test nojit tests please
@dotnet-bot test legacy tests please

**Comment by Cellule** [comment]
@dotnet-bot test slow tests please
@dotnet-bot test linux tests please
@dotnet-bot test nojit tests please
@dotnet-bot test legacy tests please

File: lib/WasmReader/Chakra.WasmReader.vcxproj [hunk 1]
```diff
@@ -0,0 +1,71 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (obastemur) [inline lib/WasmReader/Chakra.WasmReader.vcxproj]: Do we have a Github issue / todo comment etc. that reminds us to implement this on cross-platform?

**Comment by Cellule** [comment]
@dotnet-bot test slow tests please
@dotnet-bot test nojit tests please
@dotnet-bot test legacy tests please

**Comment by Cellule** [comment]
@dotnet-bot test Ubuntu ubuntu_linux_release_static please

**Pull request merged**